src/org/exist/storage/XQueryPool.java

```diff
@@ -58,6 +56,8 @@
 
     private final static Logger LOG = LogManager.getLogger(XQueryPool.class);
 
+    private ConcurrentMap<Source, Deque<CompiledXQuery>> pool = new ConcurrentHashMap<>();
+
     private long lastTimeOutCheck;
     private long lastTimeOfCleanup;
 
```

Comment:
Access to `lastTimeOutCheck` and `lastTimeOfCleanup` were previously `synchronized` but now are not. What is the concurrency guarantee of these? I think these are no longer thread safe and most likely need strong concurrency guarantees.